e2e/cypress/integration/messaging/markdown_spec.js

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @messaging @markdown
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+const testCases = [
+    {name: 'Markdown - typescript', fileKey: 'markdown_typescript'},
+    {name: 'Markdown - postgres', fileKey: 'markdown_postgres'},
+    {name: 'Markdown - latex', fileKey: 'markdown_latex'},
+    {name: 'Markdown - python', fileKey: 'markdown_python'},
+    {name: 'Markdown - shell', fileKey: 'markdown_shell'},
+];
+
+describe('Markdown', () => {
+    let townsquareLink;
+
+    before(() => {
+        // # Enable latex
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                EnableLatex: true,
+                EnableTesting: true,
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup({loginAfter: true}).then(({team}) => {
+            townsquareLink = `/${team.name}/channels/town-square`;
+            cy.visit(townsquareLink);
+        });
+    });
+
+    it('MM-T1734: Code highlighting', () => {
+        testCases.forEach((testCase) => {
```

Comment:
Each step in the TM4J test case should exactly match so that each step could be labeled (pass/fail) correctly. Each step can be identified/mapped by `'[tm4j_id]_[step_number] title'` With that, please change to,
```suggestion
    testCases.forEach((testCase, i) => {
        it(`MM-T1734_${i + 1}: Code highlighting - ${testCase.name}`, () => {
```